File: app/src/main/java/io/github/zwlxt/adbwireless/WirelessAdbUtil.java
package io.github.zwlxt.adbwireless;


public class WirelessAdbUtil {

    public static int STATUS_DISABLED = 0;
    public static int STATUS_LISTEN = 0x01;
    public static int STATUS_ACTIVE = 0x10;

    private static final String TAG = "WirelessAdbUtil";
    private static int port = 5555;

    public static boolean setEnable(boolean enabled) {
        if (enabled) {
            return ShellUtils.execute(
                    "setprop service.adb.tcp.port " + port,
                    "stop adbd",
                    "start adbd");
        } else {
            return ShellUtils.execute(
                    "setprop service.adb.tcp.port -1",
                    "stop adbd",
                    "start adbd");
        }
    }

    public static int getStatus() {
        String output = ShellUtils.executeForOutput("sh -c 'netstat -tln | grep :" + port + "'");
        if (output == null) {
            return STATUS_DISABLED;
        }

        int result = 0;
        if (output.contains("LISTEN")) {
            result |= STATUS_LISTEN;
        }

        if (output.contains("ESTABLISHED")) {
            result |= STATUS_ACTIVE;
        }

        return result;
    }

    public static void setPort(int port) {
        if (port > 0) {
            WirelessAdbUtil.port = port;
        }
    }

    public static int getPort() {
        String output = ShellUtils.executeForOutput("getprop service.adb.tcp.port");
        try {
            if (output == null) {
                return 0;
            }
            port = Integer.parseInt(output);
            return port;
        } catch (NumberFormatException e) {
            e.printStackTrace();
            return 0;
        }
    }
}

File: app/src/main/java/io/github/zwlxt/adbwireless/util/WirelessAdbUtil.kt
package io.github.zwlxt.adbwireless.util


object WirelessAdbUtil {

    var STATUS_DISABLED = 0
    var STATUS_LISTEN = 0x01
    var STATUS_ACTIVE = 0x10

    private val TAG = "WirelessAdbUtil"
    private var port = 5555

    val status: Int
        get() {
            val output = ShellUtil.executeForOutput("sh -c \"netstat -tln | grep ':$port '\"")
                    ?: return STATUS_DISABLED

            var result = 0
            if (output.contains("LISTEN")) {
                result = result or STATUS_LISTEN
            }

            if (output.contains("ESTABLISHED")) {
                result = result or STATUS_ACTIVE
            }

            return result
        }

    fun setEnable(enabled: Boolean): Boolean {
        return if (enabled) {
            ShellUtil.execute(
                    "setprop service.adb.tcp.port $port",
                    "stop adbd",
                    "start adbd")
        } else {
            ShellUtil.execute(
                    "setprop service.adb.tcp.port -1",
                    "stop adbd",
                    "start adbd")
        }
    }

    fun setPort(port: Int) {
        if (port > 0) {
            WirelessAdbUtil.port = port
        }
    }

    fun getPort(): Int {
        val output = ShellUtil.executeForOutput("getprop service.adb.tcp.port")
        try {
            if (output == null) {
                return 0
            }
            port = Integer.parseInt(output)
            return port
        } catch (e: NumberFormatException) {
            e.printStackTrace()
            return 0
        }

    }
}
